swarm/network/stream/snapshot_sync_test.go

```diff
@@ -0,0 +1,762 @@
+//on developer machines
+/*
+func TestSyncing_4_256(t *testing.T)   { testSyncing(t, 4, 256) }
+func TestSyncing_8_256(t *testing.T)   { testSyncing(t, 8, 256) }
+func TestSyncing_32_256(t *testing.T)  { testSyncing(t, 32, 256) }
+func TestSyncing_128_128(t *testing.T) { testSyncing(t, 128, 128) }
+func TestSyncing_128_256(t *testing.T) { testSyncing(t, 128, 256) }
+func TestSyncing_256_256(t *testing.T) { testSyncing(t, 256, 256) }
+func TestSyncing_256_128(t *testing.T) { testSyncing(t, 256, 128) }
+func TestSyncing_1024_64(t *testing.T) { testSyncing(t, 1024, 64) }
+func TestSyncing_1024_128(t *testing.T) { testSyncing(t, 1024, 128) }
+func TestSyncing_1024_256(t *testing.T) { testSyncing(t, 1024, 256) }
+*/
+
+func testSyncing(t *testing.T, chunkCount int, nodeCount int) {
+	ids = make([]discover.NodeID, nodeCount)
+
+	//test live and NO history
+	log.Info("Testing live and no history")
+	live = true
+	history = false
+	err := runSyncTest(chunkCount, nodeCount, live, history)
+	if err != nil {
+		t.Fatal(err)
+	}
+	//finaly test history only
+	log.Info("Testing history only")
+	live = false
+	history = true
+	err = runSyncTest(chunkCount, nodeCount, live, history)
+	if err != nil {
+		t.Fatal(err)
+	}
+	//test live and history
+	log.Info("Testing live and history")
+	live = true
+	err = runSyncTest(chunkCount, nodeCount, live, history)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+}
+
+/*
+The test generates the given number of chunks,
+then uploads these to a random node.
+Afterwards for every chunk generated, the nearest node addresses
+are identified, syncing is started, and finally we verify
+that the nodes closer to the chunk addresses actually do have
+the chunks in their local stores.
+
+The test loads a snapshot file to construct the swarm network,
+assuming that the snapshot file identifies a healthy
+kademlia network. The snapshot should have 'streamer' in its service list.
+
+This tests LIVE syncing, as the file is uploaded *after* sync streams have been setup.
+For HISTORY syncing a different test is needed.
+*/
+func runSyncTest(chunkCount int, nodeCount int, live bool, history bool) error {
+	//initialize the test struct
+	conf = &synctestConfig{}
+	//mapping of nearest node addresses for chunk hashes
+	conf.retrievalMap = make(map[string]map[string]time.Duration)
+	//map of discover ID to indexes of chunks expected at that ID
+	conf.idToChunksMap = make(map[discover.NodeID][]int)
+	//map of discover ID to kademlia overlay address
+	conf.idToAddrMap = make(map[discover.NodeID][]byte)
+	//map of overlay address to discover ID
+	conf.addrToIdMap = make(map[string]discover.NodeID)
+	conf.chunks = make([]storage.Key, 0)
+	//First load the snapshot from the file
+	trigger := make(chan discover.NodeID)
+	// channel to signal simulation initialisation with action call complete
+	// or node disconnections
+	disconnectC := make(chan error)
+	quitC := make(chan struct{})
```

Comment:
what is this used for?

Reply:
It's a parameter for the `streamTesting.WatchDisconnections()` call